TeamCode/src/main/java/org/firstinspires/ftc/teamcode/api/Robot.java

```diff
@@ -2,73 +2,150 @@
 
 import androidx.annotation.NonNull;
 
+import com.qualcomm.hardware.bosch.BNO055IMU;
 import com.qualcomm.robotcore.eventloop.opmode.OpMode;
+import com.qualcomm.robotcore.hardware.ColorSensor;
 import com.qualcomm.robotcore.hardware.DcMotor;
-import com.qualcomm.robotcore.hardware.DcMotorSimple;
+import com.qualcomm.robotcore.hardware.DcMotorEx;
+import com.qualcomm.robotcore.hardware.NormalizedColorSensor;
+import com.qualcomm.robotcore.hardware.Servo;
 
+import org.firstinspires.ftc.robotcore.external.hardware.camera.WebcamName;
 import org.firstinspires.ftc.teamcode.api.bp.AbstractRobot;
-import org.firstinspires.ftc.teamcode.api.vars.Hardware;
+import org.firstinspires.ftc.teamcode.api.bp.ArmRobot;
+import org.firstinspires.ftc.teamcode.api.bp.WheeledRobot;
+import org.firstinspires.ftc.teamcode.api.config.Constants;
+import org.firstinspires.ftc.teamcode.api.config.Naming;
+import org.firstinspires.ftc.teamcode.api.hw.Gyroscope;
+import org.firstinspires.ftc.teamcode.api.hw.SmartColorSensor;
+import org.firstinspires.ftc.teamcode.api.hw.SmartMotor;
+import org.firstinspires.ftc.teamcode.api.hw.SmartServo;
 
-public class Robot extends AbstractRobot {
-    public DcMotor frontLeft, frontRight, backLeft, backRight, duck;
-
-    public Robot() {
-        super();
-    }
+public class Robot extends AbstractRobot implements WheeledRobot, ArmRobot {
+    // Discuss if private is better idea
+    public SmartMotor bl, br, fl, fr, duck;
+    public SmartServo armLeft, armRight, clawLeft, clawRight;
+    public SmartColorSensor parkSensor;
+    public WebcamName webcam0;
+    public Gyroscope gyro0, gyro1;
+    private OpMode opMode;
 
     public Robot(OpMode opMode) {
         super(opMode);
     }
 
+    // Wheels
     @Override
-    public void initTeleOp(@NonNull OpMode opMode) {
-        this.frontLeft = opMode.hardwareMap.get(DcMotor.class, Hardware.FRONT_LEFT);
-        this.frontRight = opMode.hardwareMap.get(DcMotor.class, Hardware.FRONT_RIGHT);
-        this.backLeft = opMode.hardwareMap.get(DcMotor.class, Hardware.BACK_LEFT);
-        this.backRight = opMode.hardwareMap.get(DcMotor.class, Hardware.BACK_RIGHT);
-        this.duck = opMode.hardwareMap.get(DcMotor.class, Hardware.DUCK);
+    public void powerWheels(double flPower, double frPower, double blPower, double brPower) {
+        this.fl.setPower(flPower);
+        this.fr.setPower(frPower);
+        this.bl.setPower(blPower);
+        this.br.setPower(brPower);
     }
 
+    // Arm
+
+    /*
+    * The actual arm base uses asynchronous code, so that
+    * will not be implemented until this gets tested
+    */
+
     @Override
-    public void moveAll(double power) {
-        this.frontLeft.setPower(power);
-        this.frontRight.setPower(power);
-        this.backLeft.setPower(power);
-        this.backRight.setPower(power);
+    public void positionClaw(double position) {
```

Comment:
Is this redundant because of the movement class?

Reply:
See above comment.